# Survey Analysis: Unravelling Consumer Awareness of Eco-Labels in Kenya: Assessing the Impact of Multiple Schemes and Exploring the KEBS Framework for Enhanced Sustainable Consumption

In this notebook we will examine and test 3 hypotheses to help us assess the impact of multiples ecolabelling schemes on consumers in Kenya

In [1]:
#import the neccesary libaries and data
import pandas as pd
import numpy as np
import seaborn as sns
data = pd.read_csv("df.csv")

In [2]:
data.drop('Timestamp',axis=1,inplace=True)
data

I confirm that I have read the information above and understand that my participation is voluntary.  \
0                                                  Yes                                                    
1                                                  Yes                                                    
2                                                  Yes                                                    
3                                                  Yes                                                    
4                                                  Yes                                                    
..                                                 ...                                                    
110                                                Yes                                                    
111                                                Yes                                                    
112                                                Yes                                                    
113                                                Yes                                                    
114                                                Yes                                                    

    Kindly enter your email address to proceed What is your age?  \
0                       susanmuia128@gmail.com             18-24   
1                     c.gichuki@alustudent.com             18-24   
2                       eddiepkhiras@gmail.com             55-64   
3                  kirangakariukilaw@gmail.com             25-34   
4                       negrilson20@gmail.com              35-44   
..                                         ...               ...   
110                     c.owino@alustudent.com             18-24   
111                    harrylloyd.lh@gmail.com             18-24   
112                       mabwapeter@gmail.com             18-24   
113                       rosesuja45@gmail.com             35-44   
114                   francissteven1@gmail.com             25-34   

    What is your Gender? Which region in Kenya do you currently reside in?  \
0                 Female                                           Nairobi   
1                 Female                                           Central   
2                   Male                                             Coast   
3                   Male                                           Nairobi   
4                   Male                                            Nyanza   
..                   ...                                               ...   
110               Female                                           Nairobi   
111                 Male                                           Nairobi   
112                  NaN                                           Nairobi   
113               Female                                             Coast   
114                 Male                                             Coast   

     How aware are you about the environmental issues?(1 being the lowest,5 being the highest)  \
0                                                    5                                           
1                                                    5                                           
2                                                    5                                           
3                                                    4                                           
4                                                    4                                           
..                                                 ...                                           
110                                                  4                                           
111                                                  4                                           
112                                                  5                             

In [21]:
data.columns

Index(['I confirm that I have read the information above and understand that my participation is voluntary.',
       'Kindly enter your email address to proceed', 'What is your age?',
       'What is your Gender?',
       'Which region in Kenya do you currently reside in?',
       'How aware are you about the environmental issues?(1 being the lowest,5 being the highest)',
       'Do you actively engage in environment-friendly practices in your daily life?',
       'I am confident in my ability to distinguish between different eco-labels when shopping for products.',
       'Have you ever encountered multiple eco-labels on similar products?',
       'How often do you find it challenging to understand the environmental implications of a product based on its eco-label?',
       'Are you familiar with the KEBS (Kenya Bureau of Standards) standard mark?',
       'How much trust do you place in products that bear the KEBS label when it comes to quality and safety?',
       'Have you intentio

# Hypothesis 1
###   Multiple eco-labelling schemes create confusion among consumers in Kenya and fail to promote awareness of eco-friendly products effectively.

**Specific Question for Testing: "How often do you find it challenging to understand the environmental implications of a product based on its eco-label?"**
Let's focus on one specific aspect of the hypothesis: the relationship between encountering multiple eco-labels and the challenge of understanding environmental implications.


*H0:There is no significant relationship between encountering multiple eco-labels and the frequency of finding it challenging to understand the environmental implications of a product based on its eco-label.*
*H1:There is a significant relationship between encountering multiple eco-labels and the frequency of finding it challenging to understand the environmental implications of a product based on its eco-label*


In [22]:
# Chi-squared testing
from scipy.stats import chi2_contingency


#Create a subset of the data from to include only the relevant questions
d1 = data[['I am confident in my ability to distinguish between different eco-labels when shopping for products.','How often do you find it challenging to understand the environmental implications of a product based on its eco-label?']]

# Create a contingency table
contingency_table = pd.crosstab(d1['I am confident in my ability to distinguish between different eco-labels when shopping for products.'], d1['How often do you find it challenging to understand the environmental implications of a product based on its eco-label?'])

# Perform the Chi-squared test
chi2, p, dof, expected = chi2_contingency(contingency_table)

# Print the results
print("Chi-squared statistic:", chi2)
print("P-value:", p)
print("Degrees of freedom:", dof)
print("Expected frequencies table:")
print(expected)


Chi-squared statistic: 23.63667614670904
P-value: 0.022782117697544485
Degrees of freedom: 12
Expected frequencies table:
[[ 9.13043478  1.52173913 14.60869565  9.73913043]
 [ 4.69565217  0.7826087   7.51304348  5.00869565]
 [ 9.91304348  1.65217391 15.86086957 10.57391304]
 [ 2.86956522  0.47826087  4.59130435  3.06086957]
 [ 3.39130435  0.56521739  5.42608696  3.6173913 ]]


# Interpretation of Chi-Squared Test Results for Hypothesis 1
Introduction:

Hypothesis 1 postulates that multiple eco-labelling schemes create confusion among consumers in Kenya and fail to promote awareness of eco-friendly products effectively. To investigate this hypothesis, we conducted a Chi-squared test to examine the relationship between consumers' confidence in distinguishing between different eco-labels and their frequency of finding it challenging to understand the environmental implications of products based on eco-labels.

Chi-squared Test Results:

Chi-squared Statistic: 23.637
P-value: 0.0228
Degrees of Freedom: 12
Expected frequencies table:
Rows: Confidence in Distinguishing Eco-Labels
Columns: Frequency of Finding Eco-Labels Challenging
Interpretation:

With a 95% confidence level, corresponding to a significance level (alpha) of 0.05, we sought to be 95% confident in our conclusions. The p-value obtained in our Chi-squared test is approximately 0.0228, which is less than 0.05. This suggests that there is enough evidence to reject the null hypothesis at the 95% confidence level.

Justification for Selected Questions:

The two questions used in this analysis are:

"I am confident in my ability to distinguish between different eco-labels when shopping for products."
"How often do you find it challenging to understand the environmental implications of a product based on its eco-label?"
Question Justification:

Question 1: This question assesses consumers' confidence in distinguishing between different eco-labels. It is pertinent to Hypothesis 1, as it directly explores consumers' perceptions and confidence regarding eco-labels, a key element in the context of eco-friendly products. Consumers' confidence can influence their behavior and choices when it comes to eco-labeled products.

Question 2: This question investigates the frequency with which consumers find it challenging to understand the environmental implications of a product based on its eco-label. It is directly related to the hypothesis, as it measures the perceived effectiveness of eco-labels in conveying environmental information. If consumers frequently find it challenging to understand these implications, it aligns with the hypothesis that eco-labeling schemes may create confusion.

Interpretation and Implications:

The statistically significant association revealed by the Chi-squared test suggests a relationship between consumers' confidence in distinguishing eco-labels and the frequency of finding it challenging to understand the environmental implications of eco-labeled products. This finding holds important implications for eco-labeling strategies and consumer awareness of eco-friendly products in Kenya.

It indicates that the effectiveness of eco-labels, in terms of consumers' comprehension and confidence, is a significant factor that should be considered in eco-labeling schemes

# Hypothesis 2
### How much trust do you place in products that bear the KEBS label when it comes to quality and safety?

#### Specific Questions for Testing:

**How much trust do you place in products that bear the KEBS label when it comes to quality and safety?**

**Have you intentionally purchased products with the KEBS label because of its perceived quality and safety standards?**

Null Hypothesis (H0): There is no significant relationship between respondents' trust in products with the KEBS label and their likelihood to intentionally purchase products with the KEBS label.

Alternative Hypothesis (H1): There is a significant relationship between respondents' trust in products with the KEBS label and their likelihood to intentionally purchase products with the KEBS label

In [23]:
d2 = data[['How much trust do you place in products that bear the KEBS label when it comes to quality and safety?','Have you intentionally purchased products with the KEBS label because of its perceived quality and safety standards?']]
d2

How much trust do you place in products that bear the KEBS label when it comes to quality and safety?  \
0                                                 Some                                                      
1                                         A grest deal                                                      
2                                         A grest deal                                                      
3                                               Little                                                      
4                                               Little                                                      
..                                                 ...                                                      
110                                       A great deal                                                      
111                                       A great deal                                                      
112                                       A great deal                                                      
113                                       A great deal                                                      
114                                       A great deal                                                      

    Have you intentionally purchased products with the KEBS label because of its perceived quality and safety standards?  
0                                                  Yes                                                                    
1                                                  Yes                                                                    
2                                                  Yes                                                                    
3                                                   No                                                                    
4                                                   No                                                                    
..                                                 ...                                                                    
110                                                Yes                                                                    
111                                                Yes                                                                    
112                                                Yes                                                                    
113                                                Yes                                                                    
114                                                Yes                                                                    

[115 rows x 2 columns]

In [24]:
from scipy.stats import chi2_contingency

# Create a contingency table
contingency_table = pd.crosstab(d2['How much trust do you place in products that bear the KEBS label when it comes to quality and safety?'], d2['Have you intentionally purchased products with the KEBS label because of its perceived quality and safety standards?'])

# Perform the Chi-squared test
chi2, p, dof, expected = chi2_contingency(contingency_table)

# Print the results
print("Chi-squared statistic:", chi2)
print("P-value:", p)
print("Degrees of freedom:", dof)
print("Expected frequencies table:")
print(expected)


Chi-squared statistic: 12.000777211421386
P-value: 0.017345486628878234
Degrees of freedom: 4
Expected frequencies table:
[[ 5.63809524 31.36190476]
 [ 0.3047619   1.6952381 ]
 [ 4.72380952 26.27619048]
 [ 0.15238095  0.84761905]
 [ 5.18095238 28.81904762]]


# Interpretation of Chi-Squared Test Results for Hypothesis 2 at 95% Confidence
Introduction:

Hypothesis 2 focuses on consumer trust and preference for products bearing the KEBS label in Kenya, particularly concerning quality and safety standards. The Chi-squared test was conducted to explore the relationship between respondents' intentional purchases of products with the KEBS label due to perceived quality and safety standards and their likelihood of choosing products with the KEBS label over similar products without it.

Chi-squared Test Results:

Chi-squared Statistic: 12.001
P-value: 0.0173
Degrees of Freedom: 4
Expected frequencies table:
Rows: Intentional Purchase of KEBS-Labeled Products
Columns: Likelihood of Choosing KEBS-Labeled Products
Interpretation:

At a 95% confidence level, corresponding to a significance level (alpha) of 0.05, the p-value obtained is approximately 0.0173, which is less than 0.05. This suggests that there is enough evidence to reject the null hypothesis at the 95% confidence level.

Justification for Selected Questions:

The two questions used in this analysis are:

"Have you intentionally purchased products with the KEBS label because of its perceived quality and safety standards?"
"How likely are you to choose a product with the KEBS label over a similar product without it?"
Question Justification:

Question 1: This question directly addresses the intention to purchase products with the KEBS label based on perceived quality and safety standards. It is relevant for Hypothesis 2 as it examines consumers' actual purchasing decisions and their trust in the KEBS label for quality and safety. Trust in quality and safety is a critical aspect of consumer preferences.

Question 2: This question explores consumers' likelihood to choose products with the KEBS label over similar alternatives. It is pertinent for Hypothesis 2 as it assesses future consumer behavior and preferences. Consumers' likelihood to choose KEBS-labeled products reflects their trust and preference for such products.

Interpretation and Implications:

The Chi-squared test results indicate a statistically significant association between respondents' intentional purchases of KEBS-labeled products due to quality and safety standards and their likelihood to choose products with the KEBS label over similar alternatives. This finding has significant implications for understanding consumer trust and preference for KEBS-labeled products.

It suggests that consumers who intentionally purchase products with the KEBS label due to quality and safety standards are also more likely to continue choosing such products in the future. This reflects a strong level of trust in the KEBS label for quality assurance and safety

# Hypothesis 3: To what extent do you associate the KEBS label with government oversight and regulation?


Specific Questions for Testing:

"To what extent do you associate the KEBS label with government oversight and regulation?"

"Do you believe that products with the KEBS label are more reliable and safer than those without it?"

Null Hypothesis (H0): There is no significant relationship between respondents' association of the KEBS label with government oversight and regulation and their belief in the reliability and safety of products with the KEBS label.

Alternative Hypothesis (H1): There is a significant relationship between respondents' association of the KEBS label with government oversight and regulation and their belief in the reliability and safety of products with the KEBS label.

In [25]:
#Select the relevant columns
d3 = data[['To what extent do you associate the KEBS label with government oversight and regulation?','Do you believe that products with the KEBS label are more reliable and safer than those without it?']]
d3

To what extent do you associate the KEBS label with government oversight and regulation?  \
0                                              Neutral                                         
1                                             Strongly                                         
2                                             Strongly                                         
3                                              Neutral                                         
4                                             Somewhat                                         
..                                                 ...                                         
110                                           Strongly                                         
111                                           Strongly                                         
112                                           Somewhat                                         
113                                           Somewhat                                         
114                                           Strongly                                         

    Do you believe that products with the KEBS label are more reliable and safer than those without it?  
0                                                  Yes                                                   
1                                                  Yes                                                   
2                                                  Yes                                                   
3                                                   No                                                   
4                                                   No                                                   
..                                                 ...                                                   
110                                                Yes                                                   
111                                                Yes                                                   
112                                                Yes                                                   
113                                                Yes                                                   
114                                                Yes                                                   

[115 rows x 2 columns]

In [26]:

# Create a contingency table
contingency_table = pd.crosstab(d3['To what extent do you associate the KEBS label with government oversight and regulation?'], d3['Do you believe that products with the KEBS label are more reliable and safer than those without it?'])

# Perform the Chi-squared test
chi2, p, dof, expected = chi2_contingency(contingency_table)

# Print the results
print("Chi-squared statistic:", chi2)
print("P-value:", p)
print("Degrees of freedom:", dof)
print("Expected frequencies table:")
print(expected)


Chi-squared statistic: 49.326039283045546
P-value: 4.992228082458918e-10
Degrees of freedom: 4
Expected frequencies table:
[[ 5.63478261 12.36521739]
 [ 0.93913043  2.06086957]
 [ 4.06956522  8.93043478]
 [ 9.70434783 21.29565217]
 [15.65217391 34.34782609]]


# Interpretation of Chi-Squared Test Results for Hypothesis 3 at 95% Confidence
Introduction:

Hypothesis 3 is centered around the extent to which consumers in Kenya associate the KEBS label with government oversight and regulation. The Chi-squared test was conducted to explore the relationship between respondents' perceptions of the KEBS label's association with government oversight and their belief that products with the KEBS label are more reliable and safer than those without it.

Chi-squared Test Results:

Chi-squared Statistic: 49.326
P-value: 4.992e-10 (approximately 0.0000000004992)
Degrees of Freedom: 4
Expected frequencies table:
Rows: Association of KEBS Label with Government Oversight
Columns: Belief in Reliability and Safety of KEBS-Labeled Products
Interpretation:

At a 95% confidence level, corresponding to a significance level (alpha) of 0.05, the p-value obtained is approximately 0.0000000004992, which is much less than 0.05. This exceptionally low p-value indicates a highly significant result. Therefore, there is compelling evidence to reject the null hypothesis at the 95% confidence level.

Justification for Selected Questions:

The two questions used in this analysis are:

"To what extent do you associate the KEBS label with government oversight and regulation?"
"Do you believe that products with the KEBS label are more reliable and safer than those without it?"
Question Justification:

Question 1: This question directly addresses consumers' associations of the KEBS label with government oversight and regulation. It is highly relevant for Hypothesis 3 as it delves into perceptions and beliefs related to the role of government in product regulation. The extent of this association is vital in understanding consumer attitudes and trust.

Question 2: This question investigates consumers' beliefs about the reliability and safety of products with the KEBS label. It is directly aligned with Hypothesis 3 as it gauges consumer perceptions of the KEBS label's efficacy in ensuring product reliability and safety. These beliefs are crucial factors in consumer preferences.

Interpretation and Implications:

The Chi-squared test results indicate a highly significant association between respondents' associations of the KEBS label with government oversight and their belief that products with the KEBS label are more reliable and safer. This finding carries substantial implications for understanding consumer trust in the KEBS label and its role in product regulation.

It suggests that consumers who strongly associate the KEBS label with government oversight are also more likely to believe in the reliability and safety of products bearing the label. This association demonstrates that the KEBS label plays a role in building trust and perceptions of reliability and safety among consumers.

In the context of Hypothesis 3, these results support the idea that consumers view the KEBS label as a symbol of government-backed oversight and associate it with product reliability and safety. This trust in the label's regulatory function influences their beliefs about the safety and reliability of KEBS-labeled products

# Justification for Using the Chi-squared Test:

The Chi-squared test is suitable for your hypotheses for the following reasons:

Categorical Data: The survey responses are largely categorical in nature. The Chi-squared test is designed for the analysis of categorical data and can assess associations between categorical variables.

Independence Testing: The Chi-squared test assesses the independence of two categorical variables. It's particularly well-suited for testing whether there's a significant relationship between the categorical variables in your hypotheses.

Hypothesis-Driven: Your hypotheses involve relationships between categorical variables, making the Chi-squared test a hypothesis-driven choice.

Chi-squared Test of Independence: For your hypotheses related to associations and relationships among categorical variables, the Chi-squared test of independence is a commonly used and straightforward statistical test.

While other tests can be valuable for different types of data and hypotheses, the Chi-squared test aligns with the nature of your survey questions and the specific relationships you want to investigat

# Confirmatory Factory Analysis

In [3]:
d4 = data.copy()
d4

I confirm that I have read the information above and understand that my participation is voluntary.  \
0                                                  Yes                                                    
1                                                  Yes                                                    
2                                                  Yes                                                    
3                                                  Yes                                                    
4                                                  Yes                                                    
..                                                 ...                                                    
110                                                Yes                                                    
111                                                Yes                                                    
112                                                Yes                                                    
113                                                Yes                                                    
114                                                Yes                                                    

    Kindly enter your email address to proceed What is your age?  \
0                       susanmuia128@gmail.com             18-24   
1                     c.gichuki@alustudent.com             18-24   
2                       eddiepkhiras@gmail.com             55-64   
3                  kirangakariukilaw@gmail.com             25-34   
4                       negrilson20@gmail.com              35-44   
..                                         ...               ...   
110                     c.owino@alustudent.com             18-24   
111                    harrylloyd.lh@gmail.com             18-24   
112                       mabwapeter@gmail.com             18-24   
113                       rosesuja45@gmail.com             35-44   
114                   francissteven1@gmail.com             25-34   

    What is your Gender? Which region in Kenya do you currently reside in?  \
0                 Female                                           Nairobi   
1                 Female                                           Central   
2                   Male                                             Coast   
3                   Male                                           Nairobi   
4                   Male                                            Nyanza   
..                   ...                                               ...   
110               Female                                           Nairobi   
111                 Male                                           Nairobi   
112                  NaN                                           Nairobi   
113               Female                                             Coast   
114                 Male                                             Coast   

     How aware are you about the environmental issues?(1 being the lowest,5 being the highest)  \
0                                                    5                                           
1                                                    5                                           
2                                                    5                                           
3                                                    4                                           
4                                                    4                                           
..                                                 ...                                           
110                                                  4                                           
111                                                  4                                           
112                                                  5                             

In [5]:
d4.drop(['I confirm that I have read the information above and understand that my participation is voluntary.','Kindly enter your email address to proceed'],axis=1,inplace=True)
d4

What is your age? What is your Gender?  \
0               18-24               Female   
1               18-24               Female   
2               55-64                 Male   
3               25-34                 Male   
4               35-44                 Male   
..                ...                  ...   
110             18-24               Female   
111             18-24                 Male   
112             18-24                  NaN   
113             35-44               Female   
114             25-34                 Male   

    Which region in Kenya do you currently reside in?  \
0                                             Nairobi   
1                                             Central   
2                                               Coast   
3                                             Nairobi   
4                                              Nyanza   
..                                                ...   
110                                           Nairobi   
111                                           Nairobi   
112                                           Nairobi   
113                                             Coast   
114                                             Coast   

     How aware are you about the environmental issues?(1 being the lowest,5 being the highest)  \
0                                                    5                                           
1                                                    5                                           
2                                                    5                                           
3                                                    4                                           
4                                                    4                                           
..                                                 ...                                           
110                                                  4                                           
111                                                  4                                           
112                                                  5                                           
113                                                  3                                           
114                                                  4                                           

    Do you actively engage in environment-friendly practices in your daily life?  \
0                                                   No                             
1                                                  Yes                             
2                                                  Yes                             
3                                                  Yes                             
4                                                  Yes                             
..                                                 ...                             
110                                                Yes                             
111                                                Yes                             
112                                                Yes                             
113                                                Yes                             
114                                                Yes                             

    I am confident in my ability to distinguish between different eco-labels when shopping for products.  \
0                                       Strongly Agree                                                     
1                                                Agree                                                     
2                                    Strongly Disagree                                                     
3                                              Neutral                                                     
4                                       

In [6]:
unique_categories = {}
for column in d4.columns:
    unique_categories[column] = d4[column].unique()

print(unique_categories)

{'What is your age?': array(['18-24', '55-64', '25-34', '35-44', nan, '45-54'], dtype=object), 'What is your Gender?': array(['Female', 'Male', nan, 'Prefer not to say'], dtype=object), 'Which region in Kenya do you currently reside in?': array(['Nairobi', 'Central', 'Coast', 'Nyanza', 'Western', 'Rift Valley',
       'North Eastern'], dtype=object), 'How aware are you about the environmental issues?(1 being the lowest,5 being the highest)': array([5, 4, 3, 2, 1], dtype=int64), 'Do you actively engage in environment-friendly practices in your daily life?': array(['No', 'Yes', nan], dtype=object), 'I am confident in my ability to distinguish between different eco-labels when shopping for products.': array(['Strongly Agree', 'Agree', 'Strongly Disagree', 'Neutral',
       'Disagree'], dtype=object), 'Have you ever encountered multiple eco-labels on similar products?': array(['Yes', 'No'], dtype=object), 'How often do you find it challenging to understand the environmental implications of

In [ ]:
binary_mapping = {'Yes':1,'No':0}
categorical_mapping = {
    'Strongly Agree': 5,
    'Agree': 4,
    'Neutral': 3,
    'Disagree': 2,
    'Strongly Disagree': 1
}

In [9]:
d5 = d4.copy()

# Define mapping dictionaries for all columns
mapping_dict = {
    'What is your age?': None,  # Leave as is
    'What is your Gender?': {
        'Female': 1,
        'Male': 0,
        'Prefer not to say': 1  # Assumed to be binary
    },
    'Which region in Kenya do you currently reside in?': {
        'Nairobi': 5,
        'Central': 4,
        'Coast': 3,
        'Nyanza': 2,
        'Western': 1,
        'Rift Valley': 1,
        'North Eastern': 1
    },
    'How aware are you about the environmental issues?(1 being the lowest,5 being the highest)': {
        5: 5,
        4: 4,
        3: 3,
        2: 2,
        1: 1
    },
    'Do you actively engage in environment-friendly practices in your daily life?': {
        'Yes': 1,
        'No': 0
    },
    'I am confident in my ability to distinguish between different eco-labels when shopping for products.': {
        'Strongly Agree': 5,
        'Agree': 4,
        'Neutral': 3,
        'Disagree': 2,
        'Strongly Disagree': 1
    },
    'Have you ever encountered multiple eco-labels on similar products?': {
        'Yes': 1,
        'No': 0
    },
    'How often do you find it challenging to understand the environmental implications of a product based on its eco-label?': {
        'Frequently': 5,
        'Rarely': 4,
        'Occasionally': 3,
        'Never': 2
    },
    'Are you familiar with the KEBS (Kenya Bureau of Standards) standard mark?': {
        'Yes': 1,
        'No': 0
    },
    'How much trust do you place in products that bear the KEBS label when it comes to quality and safety?': {
        'A great deal': 5,
        'Some': 4,
        'Little': 3,
        'NaN': 1,  # Assumed to be 1 if the value is missing
        'A great deal': 5,
        'Somewhat': 2
    },
    'Have you intentionally purchased products with the KEBS label because of its perceived quality and safety standards?': {
        'Yes': 1,
        'No': 0
    },
    'To what extent do you associate the KEBS label with government oversight and regulation?': {
        'Strongly': 5,
        'Somewhat': 4,
        'Neutral': 3,
        'Not very': 2,
        'Not at all': 1
    },
    'Do you believe that products with the KEBS label are more reliable and safer than those without it?': {
        'Yes': 1,
        'No': 0
    },
    'How likely are you to choose a product with the KEBS label over a similar product without it?': {
        'Very likely': 5,
        'Somewhat likely': 4,
        'Neutral': 3,
        'Somewhat unlikely': 2,
        'Very unlikely': 1
    }
}

# Apply mapping to the DataFrame
for column, mapping in mapping_dict.items():
    if mapping:
        d5[column] = d5[column].map(mapping)

# Print the DataFrame with mappings applied
d5


What is your age?  What is your Gender?  \
0               18-24                   1.0   
1               18-24                   1.0   
2               55-64                   0.0   
3               25-34                   0.0   
4               35-44                   0.0   
..                ...                   ...   
110             18-24                   1.0   
111             18-24                   0.0   
112             18-24                   NaN   
113             35-44                   1.0   
114             25-34                   0.0   

     Which region in Kenya do you currently reside in?  \
0                                                    5   
1                                                    4   
2                                                    3   
3                                                    5   
4                                                    2   
..                                                 ...   
110                                                  5   
111                                                  5   
112                                                  5   
113                                                  3   
114                                                  3   

     How aware are you about the environmental issues?(1 being the lowest,5 being the highest)  \
0                                                    5                                           
1                                                    5                                           
2                                                    5                                           
3                                                    4                                           
4                                                    4                                           
..                                                 ...                                           
110                                                  4                                           
111                                                  4                                           
112                                                  5                                           
113                                                  3                                           
114                                                  4                                           

     Do you actively engage in environment-friendly practices in your daily life?  \
0                                                  0.0                              
1                                                  1.0                              
2                                                  1.0                              
3                                                  1.0                              
4                                                  1.0                              
..                                                 ...                              
110                                                1.0                              
111                                                1.0                              
112                                                1.0                              
113                                                1.0                              
114                                                1.0                              

     I am confident in my ability to distinguish between different eco-labels when shopping for products.  \
0                                                    5                                                      
1                                                    4                                                      
2                                                    1                                                      
3                                                    3                                                      

In [11]:
d6 = d5[['I am confident in my ability to distinguish between different eco-labels when shopping for products.','Have you ever encountered multiple eco-labels on similar products?','How often do you find it challenging to understand the environmental implications of a product based on its eco-label?']]
d6

I am confident in my ability to distinguish between different eco-labels when shopping for products.  \
0                                                    5                                                      
1                                                    4                                                      
2                                                    1                                                      
3                                                    3                                                      
4                                                    3                                                      
..                                                 ...                                                      
110                                                  3                                                      
111                                                  3                                                      
112                                                  5                                                      
113                                                  2                                                      
114                                                  4                                                      

     Have you ever encountered multiple eco-labels on similar products?  \
0                                                    1                    
1                                                    0                    
2                                                    0                    
3                                                    1                    
4                                                    1                    
..                                                 ...                    
110                                                  0                    
111                                                  0                    
112                                                  1                    
113                                                  1                    
114                                                  1                    

     How often do you find it challenging to understand the environmental implications of a product based on its eco-label?  
0                                                    5                                                                       
1                                                    4                                                                       
2                                                    5                                                                       
3                                                    4                                                                       
4                                                    3                                                                       
..                                                 ...                                                                       
110                                                  4                                                                       
111                                                  3                                                                       
112                                                  2                                                                       
113                                                  3                                                                       
114                                                  4                                                                       

[115 rows x 3 columns]

In [13]:
d6.rename({
    'I am confident in my ability to distinguish between different eco-labels when shopping for products.':'Confidence','Have you ever encountered multiple eco-labels on similar products?':'Encounter_Multiple_Labels','How often do you find it challenging to understand the environmental implications of a product based on its eco-label?':'Challenging_Understanding'
},axis=1,inplace=True)
d6

C:\Users\David\AppData\Local\Temp\ipykernel_3620\2997108162.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  d6.rename({


Confidence  Encounter_Multiple_Labels  Challenging_Understanding
0             5                          1                          5
1             4                          0                          4
2             1                          0                          5
3             3                          1                          4
4             3                          1                          3
..          ...                        ...                        ...
110           3                          0                          4
111           3                          0                          3
112           5                          1                          2
113           2                          1                          3
114           4                          1                          4

[115 rows x 3 columns]

In [30]:
import semopy

model = '''
    # Measurement model
    f1 =~ Confidence
    f2 =~ Encounter_Multiple_Labels
    f3 =~ Challenging_Understanding
'''
model = semopy.Model(model)
model.fit(d6)
 

SolverResult(fun=1.1088400864167625e-07, success=True, n_it=15, x=array([ 0.56955222,  0.93710384,  0.13830387,  0.34566716,  0.11070271,
        0.09472658, -0.09435964,  0.2231061 , -0.1027481 ]), message='Optimization terminated successfully', name_method='SLSQP', name_obj='MLW')

In [33]:
res =model.fit(d6)
print(res)

Name of objective: MLW
Optimization method: SLSQP
Optimization successful.
Optimization terminated successfully
Objective value: 0.000
Number of iterations: 1
Params: 0.570 0.937 0.138 0.346 0.111 0.095 -0.094 0.223 -0.103


# Intepretation

#### factor_loadings = [0.56955222, 0.93710384, 0.13830387]
These factor loadings correspond to the three latent factors in your CFA model: "f1," "f2," and "f3." Each factor loading value represents the strength of the relationship between the corresponding latent factor and the observed variable.

Factor f1 has a loading of approximately 0.570 on the variable "Confidence." This indicates that "Confidence" is a good indicator of the underlying construct represented by factor f1. A higher loading suggests that changes in "Confidence" are strongly associated with changes in factor f1.

Factor f2 has a loading of approximately 0.937 on the variable "Encounter_Multiple_Labels." This high loading suggests that "Encounter_Multiple_Labels" is a very strong indicator of factor f2.

Factor f3 has a loading of approximately 0.138 on the variable "Challenging_Understanding." This lower loading indicates that "Challenging_Understanding" has a weaker association with factor f3 compared to the other indicators.

It's important to note that the magnitude of the factor loading values can vary between -1 and 1, where values closer to 1 or -1 indicate a stronger relationship between the factor and the indicator. Values closer to 0 suggest a weaker relationship.

In summary, these results suggest that "Encounter_Multiple_Labels" is a strong indicator of factor f2, while "Confidence" also contributes to factor f1 but with a somewhat weaker association. "Challenging_Understanding" has a relatively weaker relationship with factor f3. The CFA model has successfully estimated these factor loadings, which can be used to understand the relationships between latent constructs and observed variables in your da


In Confirmatory Factor Analysis (CFA), the factor loadings represent the relationships between latent factors (in this case, your Eco-label Perception Construct) and the observed variables (the three questions). The factor loadings are standardized coefficients that indicate how much variance in the observed variables is explained by the underlying latent construct. Let's interpret your results with respect to these three questions and the Eco-label Perception Construct:

"I am confident in my ability to distinguish between different eco-labels when shopping for products."

Factor Loading: 0.570
Interpretation: The factor loading of 0.570 suggests that this question is moderately positively associated with the Eco-label Perception Construct. In other words, individuals who report higher confidence in distinguishing between eco-labels are more likely to score higher on the Eco-label Perception Construct. This question contributes significantly to the measurement of the Eco-label Perception Construct.
"Have you ever encountered multiple eco-labels on similar products?"

Factor Loading: 0.937
Interpretation: The factor loading of 0.937 indicates a strong positive association between encountering multiple eco-labels and the Eco-label Perception Construct. Individuals who have encountered multiple eco-labels on similar products are more likely to have a higher score on the Eco-label Perception Construct. This question is a strong indicator of the construct.
"How often do you find it challenging to understand the environmental implications of a product based on its eco-label?"

Factor Loading: 0.138
Interpretation: The factor loading of 0.138 suggests that this question is weakly positively associated with the Eco-label Perception Construct. It contributes to the measurement of the construct, but to a lesser extent compared to the other two questions. Individuals who find it challenging to understand eco-labels tend to have a slightly higher score on the Eco-label Perception Construct

# Trust in KEBS Construct:

In [35]:
d5

What is your age?  What is your Gender?  \
0               18-24                   1.0   
1               18-24                   1.0   
2               55-64                   0.0   
3               25-34                   0.0   
4               35-44                   0.0   
..                ...                   ...   
110             18-24                   1.0   
111             18-24                   0.0   
112             18-24                   NaN   
113             35-44                   1.0   
114             25-34                   0.0   

     Which region in Kenya do you currently reside in?  \
0                                                    5   
1                                                    4   
2                                                    3   
3                                                    5   
4                                                    2   
..                                                 ...   
110                                                  5   
111                                                  5   
112                                                  5   
113                                                  3   
114                                                  3   

     How aware are you about the environmental issues?(1 being the lowest,5 being the highest)  \
0                                                    5                                           
1                                                    5                                           
2                                                    5                                           
3                                                    4                                           
4                                                    4                                           
..                                                 ...                                           
110                                                  4                                           
111                                                  4                                           
112                                                  5                                           
113                                                  3                                           
114                                                  4                                           

     Do you actively engage in environment-friendly practices in your daily life?  \
0                                                  0.0                              
1                                                  1.0                              
2                                                  1.0                              
3                                                  1.0                              
4                                                  1.0                              
..                                                 ...                              
110                                                1.0                              
111                                                1.0                              
112                                                1.0                              
113                                                1.0                              
114                                                1.0                              

     I am confident in my ability to distinguish between different eco-labels when shopping for products.  \
0                                                    5                                                      
1                                                    4                                                      
2                                                    1                                                      
3                                                    3                                                      

In [36]:
d7 = d5[["How much trust do you place in products that bear the KEBS label when it comes to quality and safety?",
"Have you intentionally purchased products with the KEBS label because of its perceived quality and safety standards?",
"To what extent do you associate the KEBS label with government oversight and regulation?",
"Do you believe that products with the KEBS label are more reliable and safer than those without it?",
"How likely are you to choose a product with the KEBS label over a similar product without it?"]]
d7

How much trust do you place in products that bear the KEBS label when it comes to quality and safety?  \
0                                                  4.0                                                       
1                                                  NaN                                                       
2                                                  NaN                                                       
3                                                  3.0                                                       
4                                                  3.0                                                       
..                                                 ...                                                       
110                                                5.0                                                       
111                                                5.0                                                       
112                                                5.0                                                       
113                                                5.0                                                       
114                                                5.0                                                       

     Have you intentionally purchased products with the KEBS label because of its perceived quality and safety standards?  \
0                                                    1                                                                      
1                                                    1                                                                      
2                                                    1                                                                      
3                                                    0                                                                      
4                                                    0                                                                      
..                                                 ...                                                                      
110                                                  1                                                                      
111                                                  1                                                                      
112                                                  1                                                                      
113                                                  1                                                                      
114                                                  1                                                                      

     To what extent do you associate the KEBS label with government oversight and regulation?  \
0                                                    3                                          
1                                                    5                                          
2                                                    5                                          
3                                                    3                                          
4                                                    4                                          
..                                                 ...                                          
110                                                  5                                          
111                                                  5                                          
112                                                  4                                          
113                                                  4                                          
114                                                  5                                          

     Do you believ

In [37]:
d7.fillna(0, inplace=True)
d7

C:\Users\David\AppData\Local\Temp\ipykernel_3620\1247927630.py:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  d7.fillna(0, inplace=True)


How much trust do you place in products that bear the KEBS label when it comes to quality and safety?  \
0                                                  4.0                                                       
1                                                  0.0                                                       
2                                                  0.0                                                       
3                                                  3.0                                                       
4                                                  3.0                                                       
..                                                 ...                                                       
110                                                5.0                                                       
111                                                5.0                                                       
112                                                5.0                                                       
113                                                5.0                                                       
114                                                5.0                                                       

     Have you intentionally purchased products with the KEBS label because of its perceived quality and safety standards?  \
0                                                    1                                                                      
1                                                    1                                                                      
2                                                    1                                                                      
3                                                    0                                                                      
4                                                    0                                                                      
..                                                 ...                                                                      
110                                                  1                                                                      
111                                                  1                                                                      
112                                                  1                                                                      
113                                                  1                                                                      
114                                                  1                                                                      

     To what extent do you associate the KEBS label with government oversight and regulation?  \
0                                                    3                                          
1                                                    5                                          
2                                                    5                                          
3                                                    3                                          
4                                                    4                                          
..                                                 ...                                          
110                                                  5                                          
111                                                  5                                          
112                                                  4                                          
113                                                  4                                          
114                                                  5                                          

     Do you believ

In [40]:
# Replace the column names with shorter names
d7 = d7.rename(columns={
    "How much trust do you place in products that bear the KEBS label when it comes to quality and safety?": "Trust",
    "Have you intentionally purchased products with the KEBS label because of its perceived quality and safety standards?": "Purchase",
    "To what extent do you associate the KEBS label with government oversight and regulation?": "Association",
    "Do you believe that products with the KEBS label are more reliable and safer than those without it?": "Belief",
    "How likely are you to choose a product with the KEBS label over a similar product without it?": "Likelihood"
})
d7

Trust  Purchase  Association  Belief  Likelihood
0      4.0         1            3       1           4
1      0.0         1            5       1           5
2      0.0         1            5       1           5
3      3.0         0            3       0           3
4      3.0         0            4       0           4
..     ...       ...          ...     ...         ...
110    5.0         1            5       1           5
111    5.0         1            5       1           5
112    5.0         1            4       1           5
113    5.0         1            4       1           5
114    5.0         1            5       1           5

[115 rows x 5 columns]

In [43]:
import semopy

# Define your CFA model using the shorter column names
model = '''
    # Measurement model
    Trust =~ Trust
    Purchase =~ Purchase
    Association =~ Association
    Belief =~ Belief
    Likelihood =~ Likelihood
'''
model = semopy.Model(model)
                                                                                
# Fit the CFA model to your data
model.fit(d7)
  

SolverResult(fun=inf, success=False, n_it=1, x=array([0.05, 0.05, 0.05, 0.05, 0.05]), message='Inequality constraints incompatible', name_method='SLSQP', name_obj='MLW')

In [44]:
d7

Trust  Purchase  Association  Belief  Likelihood
0      4.0         1            3       1           4
1      0.0         1            5       1           5
2      0.0         1            5       1           5
3      3.0         0            3       0           3
4      3.0         0            4       0           4
..     ...       ...          ...     ...         ...
110    5.0         1            5       1           5
111    5.0         1            5       1           5
112    5.0         1            4       1           5
113    5.0         1            4       1           5
114    5.0         1            5       1           5

[115 rows x 5 columns]